src/components/Identity/createComponent.js

```diff
@@ -35,7 +36,7 @@ export default ({
           }
         }
 
-        return handleResponseForIdSyncs(response);
+        return handleResponseForIdSyncs(response).then(noop);
```

Comment:
@ninaceban I think I understand the problem now. It seems that `processIdSyncs` is resolving the returned promise with an array of `undefined` values. I think the proper thing to do here would be to move the `.then(noop)` to inside `processIdSyncsFactory`. The reasoning is that an array of `undefined` values isn't useful outside that module, so we should be keep it from leaking out of that module. If I'm understanding this right.

Reply:
yes, thats right, that promise gets resolved with an array of `undefined`. I'll change `then(noop)` to `processIdSyncsFactory`. Thank you